### ISSUE TITLE
UI: import namespaces with dots

### ISSUE THREAD

**Issue by crusaderky** [mdhaber/marray#91]
This doesn't work on 0.0.8:
```python
from marray.jax import numpy as mjnp
```
Workaround:
```python
import marray
mjnp = marray.__getattr__("jax.numpy")
```

**Comment by mdhaber** [comment]
In the SciPy tests, I had to use `_get_namespace` due to gh-92, so I didn't notice this one. Same thing with `dask.array`, I imagine?

I've considered making `get_namespace` public again, though, for cases like this. 

```python
import jax.numpy as xp
import marray
mxp = marray.get_namespace(xp)
```

It's just more flexible than the `from marray import x` convenience syntax, and I think we need something for this and other cases we don't yet know about.

https://github.com/mdhaber/marray/issues/6#issuecomment-2525097480 didn't like the name `get_namespace`. Suggestions?

**Comment by lucascolley** [comment]
I would say `marray_namespace`, to mirror `array_namespace`.

**Comment by mdhaber** [comment]
But the meaning is pretty different. `array_namespace` accepts arrays, not a namespace.

**Comment by lucascolley** [comment]
I still prefer it to `get_namespace` even though it isn't completely analogous to `array_namespace`.

**Comment by mdhaber** [comment]
@keewis suggested `create_namespace` and `construct_namespace`. I might also propose `wrap_namespace` and `masked_namespace`. Any of those landing for you?

**Comment by lucascolley** [comment]
`masked_namespace` is probably my favourite out of those four—I like that it describes why a user would actually want to use it (to get a masked (marray) namespace). I still prefer `marray_namespace` as it is more specific, but they're quite similar.

**Comment by mdhaber** [comment]
@crusaderky @keewis would you be happy with making `masked_namespace` public to resolve this?

**Comment by crusaderky** [comment]
> [@keewis](https://github.com/keewis) suggested `create_namespace` and `construct_namespace`. I might also propose `wrap_namespace` and `masked_namespace`. Any of those landing for you?

`wrap_namespace` sounds very explicit in its intent and also reusable as a pattern from other wrappers. Strong +1 for me.

**Comment by crusaderky** [comment]
Looking at the most complicated use case I can think of, this looks pretty clean to me:

```python
from functools import partial
import dask.array as da
import marray
import numpy as np
import quantity_array

mxp = marray.wrap_namespace(np)
dxp = da.wrap_namespace(mxp)
qxp = quantity_array.wrap_namespace(dxp)

# Create a quantity_array that wraps a dask array that 
# wraps a marray (with default mask) that wraps a numpy array
a = qxp.arange(10, units="pint")

# Apply a mask. This was very painful to write and needs a better UX.
a = qxp.asarray(dxp.map_blocks(lambda x: mxp.asarray(x, mask=x > 5), a.data), units=a.units)
```

**Comment by keewis** [comment]
I'd be happy with either of these, with a slight preference towards `masked_namespace` if we don't consider how other packages would name their version of this (if it does end up being standardized then obviously we should follow that)
